docs/h5py_dataset.rst

```diff
@@ -299,6 +299,112 @@ what you requested, and nothing more.
 >>> print(data.shape)
 (80, 10)
 
+Variable-length data
+--------------------
+
+:class:`~.datasets.hdf5.H5PYDataset` also supports variable length data. Let's
+update the image features to reflect that:
+
+>>> train_image_features = [
+...     numpy.random.randint(10, size=(3, 5, 5)).astype('float32')
+...     for i in range(90)]
+>>> test_image_features = [
+...     numpy.random.randint(10, size=(3, 6, 6)).astype('float32')
+...     for i in range(10)]
+
+In this new example, training images measure 5x5 pixels whereas test images
+measure 6x6 pixels. Image sizes could also vary within the training and test
+sets; this particular arrangement is simply for illustrative purposes.
+
+First, we put the vector features and the targets inside the HDF5 file as
+before:
+
+>>> f = h5py.File('dataset.hdf5', mode='w')
+>>> f['vector_features'] = numpy.vstack(
+...     [numpy.load('train_vector_features.npy'),
+...      numpy.load('test_vector_features.npy')])
```

Comment:
Nevermind, I did not see the point of `vector_features`, but now I understand it is a different part of the data.